### PR TITLE
fix: Schema validation to make case insensitive column name comparision

### LINKED ISSUE #385: Schema validation generate wrong result if source column name is lower and target column name is upper
Hi,

I'm try to check schema validation in BigQuery source view vs target view. But in source i have lower name column and in target i have upper case name. If i run validation then i see "column<>COLUMN" and i have status FAILED :)

I made quic change in schema_validation.py and add some 2 lines and working:
```
def schema_validation_matching(source_fields, target_fields):
    """ Compare schemas between two dictionary objects """
    results = []
    # Go through each source and check if target exists and matches
    
   #----------- GW add START
    target_fields = {k.casefold(): v for k, v in target_fields.items()}
    source_fields** = {k.casefold(): v for k, v in source_fields.items()}
   #----------- GW add STOP

    for source_field_name, source_field_type in source_fields.items():
```

Please add a parameter to configuration maybe that I can choose if i want use lower or upper.
In documentation BigQuery - https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical#case_sensitivity is see something like that:
`
Category | Case Sensitive?
Table names | Yes
Column names | No
`

Thanks,
Greg

### PR DESCRIPTION
**Pull request by latika-wadhwa**
Closes #385

### CODE REVIEW

File: data_validation/schema_validation.py [hunk 1]
```diff
@@ -97,12 +97,26 @@ def execute(self):
 def schema_validation_matching(source_fields, target_fields):
     """Compare schemas between two dictionary objects"""
     results = []
+    # Apply the casefold() function to lowercase the keys of source and target
+    source_fields_casefold = dict(
```
Comment (nehanene15) [inline data_validation/schema_validation.py]: Do you need the dict()? Just the curly braces will instantiate a dictionary `{
            source_field_name.casefold(): source_field_type
            for source_field_name, source_field_type in source_fields.items()
        }`

**Review by nehanene15 — CHANGES_REQUESTED** [review]
Let's also document under [Schema Validation](https://github.com/GoogleCloudPlatform/professional-services-data-validator#schema-validations) that this will be a case insensitive column match.

File: data_validation/schema_validation.py (hunk 1 above)
Comment (latika-wadhwa) [reply data_validation/schema_validation.py]: Don't need it. Cleaned it in the next commit.

**Review by renzokuken — APPROVED** [review]
LGTM

**Review by nehanene15 — APPROVED** [review]

**Pull request merged**